packages/clarity-js/src/layout/selector.ts

```diff
@@ -24,12 +24,15 @@ export default function(input: SelectorInput, beta: boolean = false): string {
             let selector = `${prefix}${input.tag}${suffix}`;
             let classes = Constant.Class in a && a[Constant.Class].length > 0 ? a[Constant.Class].trim().split(/\s+/) : null;
             if (beta) {
-                // In beta mode, update selector to use "id" field when available
-                // The only exception is if "id" appears to be an auto generated string token, e.g. guid or a random id
+                // In beta mode, update selector to use "id" field when available. There are two exceptions:
+                // (1) if "id" appears to be an auto generated string token, e.g. guid or a random id containing digits
+                // (2) if "id" appears inside a shadow DOM, in which case we continue to prefix up to shadow DOM to prevent conflicts
+                let shadowStart = prefix.indexOf(Constant.ShadowDomTag);
```

Comment:
```suggestion
                let shadowStart = prefix.lastIndexOf(Constant.ShadowDomTag);
```

to find the last shadow DOM

Reply:
should we do the same for `PolyfillShadowDomTag`? not totally sure if needed